lib/widgets/profile.dart

```diff
@@ -0,0 +1,334 @@
+            child: Text(entry.text));
+        }
+      case CustomProfileFieldType.user:
+        // TODO(server): This is completely undocumented.  The key to
+        //   reverse-engineering it was:
+        //   https://github.com/zulip/zulip/blob/18230fcd9/static/js/settings_account.js#L247
+        final List<dynamic> userIds = jsonDecode(entry.text);
+        if (userIds.isEmpty) {
+          return const SizedBox.shrink();
+        }
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: userIds.whereType<num>()
+            .map((userId) => store.users[userId])
+            .whereType<User>()
+            .map((user) => ProfileDataTableUser(user: user)).toList());
+      case CustomProfileFieldType.date:
+      // TODO(server): The value's format is undocumented, but empirically
+      //   it's a date in ISO format, like 2000-01-01.
+      // That's readable as is, but:
+      // TODO format this date using user's locale.
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+      case CustomProfileFieldType.shortText:
+      case CustomProfileFieldType.longText:
+      case CustomProfileFieldType.pronouns:
+      case CustomProfileFieldType.unknown:
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+    }
+  }
+}
+
+class CustomProfileFieldDisplayEntry {
+  final String label;
+  final String text;
+  final CustomProfileFieldType type;
+  final String fieldData;
+
+  CustomProfileFieldDisplayEntry({required this.fieldData, required this.label, required this.text, required this.type});
+}
+
+class ProfileDataTableLink extends StatelessWidget {
+  const ProfileDataTableLink({super.key, required this.label, required this.url});
+
+  final String label;
+  final String url;
+
+  @override
+  Widget build(BuildContext context) {
+    final Uri? parsedUrl = Uri.tryParse(url);
+    if (parsedUrl == null) {
+      return Text(label);
+    }
+    return MouseRegion(
+      cursor: SystemMouseCursors.click,
+      child: GestureDetector(
+        onTap: () async {
+          await ZulipBinding.instance.launchUrl(parsedUrl);
+        },
+        child: Text(label,
+          style: TextStyle(color: const HSLColor.fromAHSL(1, 200, 1, 0.4).toColor()))));
+  }
+}
+
+class ProfileDataTableUser extends StatelessWidget {
+  const ProfileDataTableUser({super.key, required this.user});
+
+  final User user;
+
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+      onTap: () => Navigator.push(context,
+        ProfilePage.buildRoute(context: context,
+          userId: user.userId)),
+      child: Padding(
+        padding: const EdgeInsets.all(4),
+        child: Row(
+          children: [
+            Avatar(userId: user.userId, size: 32, borderRadius: 32 / 8),
+            const SizedBox(width: 8),
+            Text(user.fullName), // TODO(#196) render active status
```

Comment:
Let's add a test for a user with a long name — I expect it'll throw an error for overflow in this `Row`.